storage/sparse_merkle/src/nibble_path/mod.rs

```diff
@@ -73,6 +74,23 @@ impl NibblePath {
         self.num_nibbles += 1;
     }
 
+    /// Pops a nibble from the end of the nibble path.
+    pub fn pop(&mut self) -> Option<u8> {
+        let poped_nibble = if self.num_nibbles % 2 == 0 {
```

Comment:
I feel it's a little bit more readable if you deal with the None case (self.num_nibbles == 0) beforehand, andme make `poped_nibble.is_some()` a `popped_nibble.expect()`. But up to you.